### ISSUE TITLE
Insomnia is not converting some CURL requests to GraphQL properly

### ISSUE THREAD

**Issue by petermilan** [Kong/insomnia#1825]
- Insomnia Version: 7.0.5
- Operating System: Ubuntu 18.04.3 LTS

## Details
Insomnia is not converting some CURL requests to GraphQL properly. The endlines are kept in the GraphQL output as strings.

Steps to reproduce:
Past this CURL string into insomnia
```
curl 'http://localhost:4000/api/graphql' -H 'Connection: keep-alive' -H 'accept: */*' -H 'Origin: http://localhost:4000' -H 'authorization: Bearer d1aea2f4d2714202bd66f80feb8635d0' -H 'User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/78.0.3904.87 Safari/537.36' -H 'DNT: 1' -H 'content-type: application/json' -H 'Sec-Fetch-Site: same-origin' -H 'Sec-Fetch-Mode: cors' -H 'Referer: http://localhost:4000/test' -H 'Accept-Encoding: gzip, deflate, br' -H 'Accept-Language: sk-SK,sk;q=0.9,cs;q=0.8,en-US;q=0.7,en;q=0.6' --data-binary $'{"operationName":"myQuery","variables":{"f":{}},"query":"query myQuery($testVariable: Boolean\u0021) {\\n  myQuery(filter: $f) {\\n    data {\\n      id\\n      __typename\\n    }\\n    __typename\\n  }\\n}\\n"}' --compressed
```

Following body will be generated:
```
query myQuery($testVariable: Boolean!) {\n  myQuery(filter: $f) {\n    data {\n      id\n      __typename\n    }\n    __typename\n  }\n}\n
```

Note that the **\n** is present in the output which should not be present.

The curl was generated by **Google Chrome Version 78.0.3904.87 (Official Build) (64-bit)**

The issue is that chrome is escaping endlines with extra slash  `\\n`
I tried to generate same CURL request from Firefox which worked fine in insomnia. Difference was that the curl request in FF was containing just `\n`

![image](https://user-images.githubusercontent.com/917147/70131553-c8e17600-1682-11ea-8aed-393fcf903e6f.png)

**Comment by gschier** [comment]
Based on your description, this seems like a bug with Chrome, not Insomnia. Correct?

**Comment by gschier** [comment]
Can you also post the Curl command that Firefox generates?

**Comment by petermilan** [comment]
Well not entirely sure about the fact if it is bug in chrome, as I dont know details about CURL definition. I can only confirm that the CURL command generated by chrome is executable fine from commandline on kubuntu.

**Comment by petermilan** [comment]
regenerated similar requests using GQL  playground:

**Chrome:**
`curl 'http://localhost:4000/api/graphql' -H 'Connection: keep-alive' -H 'accept: */*' -H 'Origin: http://localhost:4000' -H 'authorization: Bearer 00000000000000000000000000000000' -H 'User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/78.0.3904.87 Safari/537.36' -H 'DNT: 1' -H 'content-type: application/json' -H 'Sec-Fetch-Site: same-origin' -H 'Sec-Fetch-Mode: cors' -H 'Referer: http://localhost:4000/api/graphql' -H 'Accept-Encoding: gzip, deflate, br' -H 'Accept-Language: sk-SK,sk;q=0.9,cs;q=0.8,en-US;q=0.7,en;q=0.6' --data-binary $'{"operationName":"myQuery","variables":{},"query":"query myQuery($testVariable: Boolean\u0021) {\\n  myQuery(filter: $f) {\\n    data {\\n      id\\n      __typename\\n    }\\n    __typename\\n  }\\n}\\n"}' --compressed`

**Firefox:**
`curl 'http://localhost:4000/api/graphql' -H 'User-Agent: Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:70.0) Gecko/20100101 Firefox/70.0' -H 'Accept: */*' -H 'Accept-Language: en-US,en;q=0.5' --compressed -H 'Referer: http://localhost:4000/api/graphql' -H 'content-type: application/json' -H 'authorization: Bearer 00000000000000000000000000000000' -H 'Origin: http://localhost:4000' -H 'Connection: keep-alive' --data '{"operationName":"myQuery","variables":{},"query":"query myQuery($testVariable: Boolean!) {\n  myQuery(filter: $f) {\n    data {\n      id\n      __typename\n    }\n    __typename\n  }\n}\n"}'`

The CURL commands are  obviously different from first look Chrome is  using some compressed flag.

**Comment by Traksewt** [comment]
It is obvious when you study both, but to save anyone else some time, I got it working by changing \\\\n to \n  and changing '--data binary $' to '--data '

**Comment by petermilan** [comment]
@Traksewt  thanks for feedback, I made me a  bash script on my kubuntu which is  taking the CURL from clipboard, replacing the characters like you described and putting back to clipboard :) so I can just paste modified curl into insomnia. Here is the bash script:

`xsel -b | sed -e "s|--data-binary \\$|--data |g"  | sed -e "s|\\\n|n|g"  | xsel -b`

perhaps it can help to someone.


`xsel -b` - is utility to work with clipboard
` sed -e "s|--data-binary \\$|--data |g"  | sed -e "s|\\\n|n|g"` is doing the trick to replace the strings

**Comment by gerrytan** [comment]
Hi I'm also affected by this problem. Just curious, anyone able to point me at the part of Insomnia code that parses that `--data-binary $'{"bunchOfGql"...` ? The curl command works fine in shell, I want to explore further if this is an improvement we can make in Insomnia

**Comment by gerrytan** [comment]
I did some investigation and I suspect we can alter the behaviour of https://github.com/Kong/insomnia/blob/develop/packages/insomnia-app/app/models/request.js#L263 to fix this

I still need to have a local environment running, place debugger etc. to confirm my understanding ;)

**Comment by gerrytan** [comment]
@gschier @petermilan let me know your thoughts on the PR I proposed to fix this (#2888) 😉
